src/Views/AppInfoView.vala

```diff
@@ -519,8 +524,35 @@ namespace AppCenter.Views {
                     });
                 });
 
-                foreach (var url in urls) {
-                    load_screenshot (url);
+                string?[] screenshot_files = new string?[urls.length ()];
+                int[] results = new int[urls.length ()];
+                int completed = 0;
+
+                // Fetch each screenshot in parallel.
+                for (int i = 0; i < urls.length (); i++) {
+                    string url = urls.nth_data (i);
+                    string? file = null;
+                    int index = i;
+
+                    cache.fetch.begin (url, (obj, res) => {
+                        results[index] = cache.fetch.end (res, out file);
+                        screenshot_files[index] = file;
+                        completed++;
+                    });
+                }
+
+                cache.maintain ();
+
+                // TODO: dynamically load screenshots as they become available.
+                while (urls.length () != completed) {
```

Comment:
I think this _could_ crash, since `completed` might be modified on two separate threads, on this one and in the cache.fetch async callback

Reply:
I don't think `begin ()` actually creates a separate thread. The operation is purely done in a synchronous manner.